### ISSUE TITLE
Add option to move a technology from one endpoint to another

### ISSUE THREAD

**Issue by antw** [quintel/etmoses#1116]
Suggested by @DorinevanderVlies 

It would be useful for users to be able to move a technology from one endpoint to another. Perhaps hidden in the "Advanced settings" section of a technology so as not to clutter the form too badly.

<img width="519" alt="screen shot 2016-06-21 at 10 52 05" src="https://cloud.githubusercontent.com/assets/4383/16225399/3756bc3e-379e-11e6-8587-71072f8a0242.png">

<img width="318" alt="screen shot 2016-06-21 at 10 52 17" src="https://cloud.githubusercontent.com/assets/4383/16225406/3d78e772-379e-11e6-8f6e-ea62c9c57b30.png">

The motivation is that a visitor may create a scenario in the ETM which is meant to represent a real region, and then wants to move technologies from one endpoint to another to more closely match the real-life situation. Currently, they would have to delete and re-create the technology on the other endpoint, making sure to get all the attributes identical. A "move to" feature would save a lot of effort.

@grdw As I'm not too familiar with the technology editor, I'm not confident in estimating how long this would take. Do you have any idea how hard this might be?

I can imagine buffers complicating things somewhat: a user might want to move a space heating technology to a node with no space heating composite; do we not give them that option – excluding that node from the list of possible destinations – or do we automatically create a space heating composite on the destination node? Not sure... the former is probably more "correct", but less intuitive.

@ChaelKruip Do you think this would be useful?

**Comment by grdw** [comment]
> As I'm not too familiar with the technology editor, I'm not confident in estimating how long this would take. Do you have any idea how hard this might be?

Well, the easy way would be a 4 hours? That's just moving the technology under a separate node and updating the matrix and making buffers move correctly.

What I think that users want to happen here is that when your matrix looks like:

````
- Endpoint 1
   EV-car (10 units)
- Endpoint 2
   EV-car (5 units)
````

Move EV-car from Endpoint 2 -> Endpoint 1:

````
- Endpoint 1
   EV-car (15 units)
- Endpoint 2
````
So should it be possible to merge? Or only merge when the capacities / volumes / demands are not similar? 

> a user might want to move a space heating technology to a node with no space heating composite

Maybe you should only have the option to move buffers and not the technologies that lie under them? Or maybe we should exclude that end-point option when an end-point doesn't have that specific buffer.

**Comment by antw** [comment]
> So should it be possible to merge? Or only merge when the capacities / volumes / demands are not similar?

That's a good point. Merging techs would be nice, but I guess the attributes would have to be identical.

> Maybe a user should only have the option to move buffers and not the technologies that lie under them? 

This also seems like a good suggestion to me, but it might be a bit drastic to move the entire buffer from one node to another (all heat demand would disappear from the source node).

@DorinevanderVlies Do you have any preference for how we would handle moving space heating and hot water technologies?

**Comment by DorinevanderVlies** [comment]
Merging techs when identical sounds like a good idea

Regarding space heating and hot water technologies. When I move my electricity demand from one node to the other (because my households are not evenly distributed) I would like (a representative portion of) my buffer to move as well.

For now (Ameland project) I will try to get around this by using .csv input for my technology table.

**Comment by antw** [comment]
I like the idea of moving technologies, but with the need to merge techs, move buffers and demand, it seems that this could be a rabbit-hole of complexity and special-cases. 😬 

Perhaps a better approach would be for us to set a number of units on the endpoints, and assign technologies in proportion *when importing from ETModel*. That is, if you have two endpoints and one has `units: 2` and the other `units: 1`, then the first endpoint should get twice the number of technologies and demand.

This is described in #336.

**Comment by grdw** [comment]
> assign technologies in proportion when importing from ETMode

Yes, that would be better. Couldn't we also allow for the whitelisting of certain technologies? Let's say endpoint-1 can only have Solar Panels or something along those lines.

**Comment by DorinevanderVlies** [comment]
The idea to influence the way the technologies are distributed after importing from ETM would be a great improvement as well. 
This could be complementary or instead of shuffling technologies around.


> Couldn't we also allow for the whitelisting of certain technologies

I agree that whitelisting  certain technologies would be nice.

>  assign technologies in proportion when importing from ETModel

Yes, if the combination of the above is implemented I think it will be easier to make the initial scenario represent a real situation

**Comment by DorinevanderVlies** [comment]
cc @dennisschoenmakers and @ChaelKruip

**Comment by ChaelKruip** [comment]
I like the following idea (courtesy of @antw): When the user creates a new LES, the current 'technology matrix interface' is not shown but the user will be looking at something like this:
![20141209_topology_interface_v1_key](https://cloud.githubusercontent.com/assets/1303760/16457091/8e0af25c-3e1a-11e6-92c4-e6d84953c661.png)
Here, the user can move technologies from one end-point to another and change concurrencies. 
After completing this, the next step would be to show the load page.

Later, we can make the technology blobs clickable so the user can edit its attributes. For now, clicking 'edit' will lead to the technology matrix as it currently is.
